src/main/java/carpet/script/CarpetExpression.java

```diff
@@ -2899,6 +2900,60 @@ private void API_AuxiliaryAspects()
             return (cc, tt) -> time;
         });
 
+        this.expr.addLazyFunction("unix_time",0,(c,t,lv)->{
+            Value time = new NumericValue(System.currentTimeMillis());
+            return (_c,_t)-> time;
+        });
+
+        this.expr.addLazyFunction("convert_date",-1,(c,t,lv)->{
+            for (int i = 0; i<lv.size();i++){
+                if(!(lv.get(i).evalValue(c) instanceof NumericValue))
+                    throw new InternalExpressionException("Attempted to pass non-numerical value to convert_date");
+            }
+
+            if(lv.size()==1) {
+                Date date = new Date(NumericValue.asNumber(lv.get(0).evalValue(c)).getLong());
+                List<Value> value_list = new ArrayList<>();
+                //l(year, month, date, hour, minute, second,day_long_name)
+                value_list.add(new NumericValue(date.getYear()+1900));//Cos returns year - 1900
+                value_list.add(new NumericValue(date.getMonth()));
+                value_list.add(new NumericValue(date.getDate()));
+                value_list.add(new NumericValue(date.getHours()));
+                value_list.add(new NumericValue(date.getMinutes()));
+                value_list.add(new NumericValue(date.getSeconds()));
+
+                switch (date.getDay()){
```

Comment:
Don't return strings. Return getDay - with the only exception that sunday is 7, not 0

Reply:
no. SUnday is 0

Reply:
wiat what?

Reply:
oh I thought iw was 0 cos I got 1 for monday

Reply:
lemme fix